File: API/WorkoutApp.API/Models/WorkoutSchedule.cs
namespace WorkoutApp.API.Models;

public class WorkoutSchedule : NamedEntity
{
    public required Guid UserId { get; set; }
    public User? User { get; set; }

    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; } = null;

    public IEnumerable<ScheduleDay> ScheduleDays { get; set; } = [];
}

File: webapp/app/models/workoutSchedule.ts
import type { NamedEntity } from "./base";
import type { ScheduleDay } from "./scheduleDay";

export interface WorkoutSchedule extends NamedEntity {
  userId: string; // GUID of owner
  startDate: Date;
  endDate?: Date | null;
  // scheduleDays will typically be loaded separately; include optional relation
  scheduleDays?: ScheduleDay[];
}
